url11fb0973cf_raichuman9_Swimmer526_tgz-pJ8-fr_inria_optimization_cmaes_examples_CMAExample1J8/src/ubc/swim/gui/SwimController.java

```diff
@@ -0,0 +1,476 @@
+					Thread.sleep(sleepTime);
+				} catch (InterruptedException ex) {
+				}
+			}
+
+			beforeTime = System.nanoTime();
+		} // end of run loop
+	}
+	
+	//////////////////////////////////////// OUTPUT RENDERING STUFF BELOW ////////////////////////////////////////
+	
+	/**
+	 * Returns BufferedImage with contents of given image
+	 * Source: http://www.exampledepot.com/egs/java.awt.image/image2buf.html
+	 * @param image
+	 * @return
+	 */
+	private static BufferedImage getBufferedImage(Image image) {
+	    if (image instanceof BufferedImage) {
+	        return (BufferedImage)image;
+	    }
+
+	    // This code ensures that all the pixels in the image are loaded
+	    image = new ImageIcon(image).getImage();
+
+	    // Determine if the image has transparent pixels; for this method's
+	    // implementation, see Determining If an Image Has Transparent Pixels
+	    boolean hasAlpha = hasAlpha(image);
+
+	    // Create a buffered image with a format that's compatible with the screen
+	    BufferedImage bimage = null;
+	    GraphicsEnvironment ge = GraphicsEnvironment.getLocalGraphicsEnvironment();
+	    try {
+	        // Determine the type of transparency of the new buffered image
+	        int transparency = Transparency.OPAQUE;
+	        if (hasAlpha) {
+	            transparency = Transparency.BITMASK;
+	        }
+
+	        // Create the buffered image
+	        GraphicsDevice gs = ge.getDefaultScreenDevice();
+	        GraphicsConfiguration gc = gs.getDefaultConfiguration();
+	        bimage = gc.createCompatibleImage(
+	            image.getWidth(null), image.getHeight(null), transparency);
+	    } catch (HeadlessException e) {
+	        // The system does not have a screen
+	    }
+
+	    if (bimage == null) {
+	        // Create a buffered image using the default color model
+	        int type = BufferedImage.TYPE_INT_RGB;
+	        if (hasAlpha) {
+	            type = BufferedImage.TYPE_INT_ARGB;
+	        }
+	        bimage = new BufferedImage(image.getWidth(null), image.getHeight(null), type);
+	    }
+
+	    // Copy image to buffered image
+	    Graphics g = bimage.createGraphics();
+
+	    // Paint the image onto the buffered image
+	    g.drawImage(image, 0, 0, null);
+	    g.dispose();
+
+	    return bimage;
+	}
+	
+	/**
+	 * Returns true if the specified image has transparent pixels
+	 * Source: http://www.exampledepot.com/egs/java.awt.image/HasAlpha.html
+	 * @param image
+	 * @return
+	 */
+	private static boolean hasAlpha(Image image) {
+	    // If buffered image, the color model is readily available
+	    if (image instanceof BufferedImage) {
+	        BufferedImage bimage = (BufferedImage)image;
+	        return bimage.getColorModel().hasAlpha();
+	    }
+
+	    // Use a pixel grabber to retrieve the image's color model;
+	    // grabbing a single pixel is usually sufficient
+	     PixelGrabber pg = new PixelGrabber(image, 0, 0, 1, 1, false);
+	    try {
+	        pg.grabPixels();
+	    } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)